File: hanjo/개인용/programmers/탐욕법/L1_체육복/solution.py
def solution(n, lost, reserve):
    # 도둑 맞았는데 여별 갖고있는 놈 미리 제거해야됨.
    for r in reserve:
        for l in lost:
            if r==l:
                lost.remove(l)
                reserve.remove(r)
    for r in reserve:
        for l in lost:
            if r+1==l or r-1==l or r==l:
                lost.remove(l)
                break
    answer = n - len(lost)            
    return answer

print(solution(5,[2, 4],[1, 3, 5]))
print(solution(5,[2, 4],[3]))
print(solution(3,[3],[1]))
print(solution(3,[3],[3]))
print(solution(5, [2,3], [3,4]))

File: hanjo/개인용/programmers/탐욕법/L1_체육복/Solution.java
package 탐욕법.L1_체육복;

class Solution {
    public static int solution(int n, int[] lost, int[] reserve) {
    	for(int i=0; i<lost.length; i++) {
    		for(int j=0; j<reserve.length; j++) {
    			if(lost[i] == reserve[j]) {
    				lost[i] = -1;
    				reserve[j] = -1;
    			}
    		}
    	}
    	for(int i=0; i<lost.length; i++) {
    		for(int j=0; j<reserve.length; j++) {
    			if(lost[i]-1 == reserve[j] || lost[i]+1 == reserve[j]) {
    				lost[i] = -1;
    				reserve[j] = -1;
    			}
    		}
    	}
    	int answer = n;
    	for(int l : lost) {
    		if(l != -1)
    			answer--;
    	}
        return answer;
    }
    
	public static void main(String[] args) {
		int n= 5;	
		int[] lost = {2, 4};
		int[] reserve = {1, 3, 5};
		System.out.println(solution(n, lost, reserve));
		
	}
}
